coldfront/core/utils/management/commands/audit_data.py

```diff
@@ -0,0 +1,323 @@
+            options['lrc_user_billing'] = True
+            options['lrc_recharge_allocation_billing'] = True
+            options['lrc_recharge_allocation_user_billing'] = True
+            options['lrc_pis_lbl_employee'] = True
+        for option in options:
+            if isinstance(options[option], bool) and options[option] and \
+                    option not in ('all', 'lrc_all'):
+                self.stdout.write(self.style.SUCCESS(f'Running {option}...'))
+                getattr(self, f'handle_{option}')()
+                self.stdout.write(self.style.SUCCESS('Audit complete.') + '\n\n')
+
+    def handle_allocation_date(self):
+        '''
+        Assert that all compute allocations have the expected start_date
+        and end_date values.
+        - Allocation end date is after the start date
+        - FCAs and PCAs:
+          FCAs and PCAs are valid within a so-called “Allowance Year”
+          (June 1st - May 31st for BRC, October 1st - September 30th for LRC):
+            - Inactive ones (ones that were not renewed) should have a start
+                date set to the start of the current allowance year
+                AllocationPeriod, but no end date.
+            - Active ones (ones that were renewed, or were created during
+                this allowance year) should have a start before and an end date
+                equal to that of the current allowance year AllocationPeriod.
+          ICAs (Instructional Computing Allowances) are valid within a
+          particular UC Berkeley semester.
+            - Inactive ones should have a start date, but no end date.
+            - Active ones (currently none on production) should match a current
+                (has started, but has not ended) instructional AllocationPeriod.
+          Recharge and Condo may/should have start dates, but don’t have end
+          dates, as they don’t end.
+        '''
+        allocations = Allocation.objects.select_related('project',
+                            'project__status').prefetch_related('resources') \
+            .filter(resources__name__endswith='Compute') \
+            .order_by('resources__name', 'project__name',
+                      'project__status__name', 'start_date', 'end_date') \
+            .values('id', 'project__status__name',
+                   'project__name','start_date', 'end_date',
+                   'resources__name')
+
+        FCA_PCA_ALLOCATION_PERIOD = get_current_allowance_year_period()
+        ICA_ALLOCATION_PERIODS = AllocationPeriod.objects.filter(
+                                    Q(end_date__gt=display_time_zone_current_date())
+                                    & (Q(name__startswith='Fall Semester')
+                                       | Q(name__startswith='Spring Semester')
+                                       | Q(name__startswith='Summer Sessions') \
+                                      )).all()
+
+        # TODO: Change these if lrc_enabled
```

Comment:
After this code was written, as part of readying the codebase for both deployments, I added `coldfront.core.resource.utils_.allowance_utils` to handle the differing allowance types across the two.

There's a function for getting the correct allowance for a `Project` object. Then, based on whether it's yearly or instructional, you can perform the right checks. Project prefixes shouldn't be hard-coded (except maybe in tests) moving forward.

Reply:
I used `ComputingAllowance(allocation).is_recharge() or ComputingAllowance(allocation).has_infinite_service_units()` to account for condo and recharge, because there's no check for condo resources.

Reply:
Actually I went ahead and added is_condo myself

Reply:
Looks good, thanks.